core/network/linklayer.go

```diff
@@ -86,3 +90,42 @@ func stringLengthBetween(value string, minLength, maxLength uint) bool {
 	length := uint(len(value))
 	return length >= minLength && length <= maxLength
 }
+
+// LinkLayerDeviceUUID represents a relation unique identifier.
+type LinkLayerDeviceUUID string
+
+// NewLinkLayerDeviceUUID is a convenience function for generating a new
+// link layer device uuid.
+func NewLinkLayerDeviceUUID() (LinkLayerDeviceUUID, error) {
+	id, err := uuid.NewUUID()
+	if err != nil {
+		return LinkLayerDeviceUUID(""), err
+	}
+	return LinkLayerDeviceUUID(id.String()), nil
+}
+
+// ParseUUID returns a new UUID from the given string. If the string is not a
+// valid uuid an error satisfying [errors.NotValid] will be returned.
+func ParseLinkLayerDeviceUUID(value string) (LinkLayerDeviceUUID, error) {
```

Comment:
The two Parse* funcs introduced in this PR are not used. Is it possible to put another PR up removing them @hmlanigan ?

It was the thing that caught my eye when seeing the notification. Specifically I was worried that we were calling these functions from the state layer when hydrating the DDL type to the core type and the potential for a bad error case to creep in.